### PR TITLE
update withNavigation doc

### PR DESCRIPTION
**Pull request by bySabi**
Updating `doc` for reflect API changes introduced on PR: https://github.com/react-navigation/react-navigation-core/pull/30#issuecomment-458306044

### CODE REVIEW

**Comment by bySabi** [comment]
@vonovak I've checked it and I do not see that another file has to update.
Please correct me is I'm wrong

**Comment by vonovak** [comment]
@bySabi does this explanation help? https://github.com/react-navigation/react-navigation.github.io/blob/source/.github/PULL_REQUEST_TEMPLATE.md

**Comment by bySabi** [comment]
**@bySabi does this explanation help? https://github.com/react-navigation/react-navigation.github.io/blob/source/.github/PULL_REQUEST_TEMPLATE.md**

I hope soo :-)

File: website/versioned_docs/version-3.x/with-navigation.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+```js
+// MyBackButton.js
+export default withNavigation(MyBackButton);
+
+// MyNavBar.js
+<MyBackButton onRef={(elem) => this.backButton = elem} />
+```
+
+- If you need to use `withNavigation` HOC as the wrapped component of a custom HOC that use `onRef` prop API too, in order to prevent `onRef` callback fuction be called twice, `onRef` injectiong must be disabled using second param `config` object.
```
Comment (vonovak) [inline website/versioned_docs/version-3.x/with-navigation.md]: After reading this over, I think this paragraph should have `### Config` heading, with a clear listing of what the config object can contain (even though there is just one option available - this is a the api reference: kinda like [here](https://reactnavigation.org/docs/en/navigation-prop.html#navigator-dependent-functions)). Does this make sense?

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
```
Comment (vonovak) [inline yarn.lock]: can you please remove this file?

**Comment by satya164** [comment]
Closing due to no response.

**Pull request closed without merging**